commit 11a80ba99fad79f05379e4143c05a262edaa159e
Author: avonazhrets <avonazhrets@gmail.com>
Date:   Mon Apr 19 00:29:29 2021 +0300

    patch18

diff --git a/libglusterfs/src/store.c b/libglusterfs/src/store.c
index e805f20..ac432cc 100644
--- a/libglusterfs/src/store.c
+++ b/libglusterfs/src/store.c
@@ -70,6 +70,9 @@ gf_store_mkstemp (gf_store_handle_t *shandle)
                         "Failed to open %s.", tmppath);
         }
 out:
+        if (!shandle) {
+                return -1;
+        }
         return shandle->tmp_fd;
 }
 
