commit 9c7febf1c952e9e2e9f9ac6744a27d731c7d6417
Author: avonazhrets <avonazhrets@gmail.com>
Date:   Mon Apr 26 15:05:21 2021 +0300

    patch34

diff --git a/xlators/mgmt/glusterd/src/glusterd-syncop.c b/xlators/mgmt/glusterd/src/glusterd-syncop.c
index 4be3d97..b63e61c 100644
--- a/xlators/mgmt/glusterd/src/glusterd-syncop.c
+++ b/xlators/mgmt/glusterd/src/glusterd-syncop.c
@@ -96,7 +96,9 @@ gd_collate_errors (struct syncargs *args, int op_ret, int op_errno,
                                 break;
                         }
                 }
-                op_err[len] = '\0';
+                if (len > 0) {
+                        op_err[len] = '\0';
+                }
 
                 if (args->errstr) {
                         len = snprintf (err_str, sizeof(err_str) - 1,
