commit acda1b352799c412b0ebf646bd125b11b2e7e82b
Author: avonazhrets <avonazhrets@gmail.com>
Date:   Fri May 14 00:49:10 2021 +0300

    patch62

diff --git a/xlators/cluster/afr/src/afr-common.c b/xlators/cluster/afr/src/afr-common.c
index 2925a1d..2db1c27 100644
--- a/xlators/cluster/afr/src/afr-common.c
+++ b/xlators/cluster/afr/src/afr-common.c
@@ -5260,10 +5260,16 @@ out:
 int
 afr_local_init (afr_local_t *local, afr_private_t *priv, int32_t *op_errno)
 {
+        int __ret = -1;
         local->op_ret = -1;
         local->op_errno = EUCLEAN;
 
-	syncbarrier_init (&local->barrier);
+        __ret = syncbarrier_init(&local->barrier);
+        if (__ret) {
+                if (op_errno)
+                        *op_errno = __ret;
+                goto out;
+        }
 
         local->child_up = GF_CALLOC (priv->child_count,
                                      sizeof (*local->child_up),
diff --git a/xlators/mgmt/glusterd/src/glusterd-mgmt.c b/xlators/mgmt/glusterd/src/glusterd-mgmt.c
index 13a4526..796667c 100644
--- a/xlators/mgmt/glusterd/src/glusterd-mgmt.c
+++ b/xlators/mgmt/glusterd/src/glusterd-mgmt.c
@@ -718,7 +718,9 @@ glusterd_mgmt_v3_initiate_lockdown (glusterd_op_t op, dict_t *dict,
 
         /* Sending mgmt_v3 lock req to other nodes in the cluster */
         gd_syncargs_init (&args, NULL);
-        synctask_barrier_init((&args));
+        ret = synctask_barrier_init((&args));
+        if (ret)
+                goto out;
         peer_cnt = 0;
 
         rcu_read_lock ();
@@ -1072,7 +1074,9 @@ glusterd_mgmt_v3_pre_validate (glusterd_op_t op, dict_t *req_dict,
 
         /* Sending Pre Validation req to other nodes in the cluster */
         gd_syncargs_init (&args, req_dict);
-        synctask_barrier_init((&args));
+        ret = synctask_barrier_init((&args));
+        if (ret)
+                goto out;
         peer_cnt = 0;
 
         rcu_read_lock ();
@@ -1358,7 +1362,9 @@ glusterd_mgmt_v3_brick_op (glusterd_op_t op, dict_t *req_dict, char **op_errstr,
 
         /* Sending brick op req to other nodes in the cluster */
         gd_syncargs_init (&args, NULL);
-        synctask_barrier_init((&args));
+        ret = synctask_barrier_init((&args));
+        if (ret)
+                goto out;
         peer_cnt = 0;
 
         rcu_read_lock ();
@@ -1624,7 +1630,9 @@ glusterd_mgmt_v3_commit (glusterd_op_t op, dict_t *op_ctx, dict_t *req_dict,
 
         /* Sending commit req to other nodes in the cluster */
         gd_syncargs_init (&args, op_ctx);
-        synctask_barrier_init((&args));
+        ret = synctask_barrier_init((&args));
+        if (ret) 
+                goto out;
         peer_cnt = 0;
 
         rcu_read_lock ();
@@ -1869,7 +1877,9 @@ glusterd_mgmt_v3_post_validate (glusterd_op_t op, int32_t op_ret, dict_t *dict,
 
         /* Sending Post Validation req to other nodes in the cluster */
         gd_syncargs_init (&args, req_dict);
-        synctask_barrier_init((&args));
+        ret = synctask_barrier_init((&args));
+        if (ret)
+                goto out;
         peer_cnt = 0;
 
         rcu_read_lock ();
@@ -2051,7 +2061,9 @@ glusterd_mgmt_v3_release_peer_locks (glusterd_op_t op, dict_t *dict,
 
         /* Sending mgmt_v3 unlock req to other nodes in the cluster */
         gd_syncargs_init (&args, NULL);
-        synctask_barrier_init((&args));
+        ret = synctask_barrier_init((&args));
+        if (ret)
+                goto out;
         peer_cnt = 0;
 
         rcu_read_lock ();
diff --git a/xlators/mgmt/glusterd/src/glusterd-snapshot.c b/xlators/mgmt/glusterd/src/glusterd-snapshot.c
index 31f4d95..fa93a8b 100644
--- a/xlators/mgmt/glusterd/src/glusterd-snapshot.c
+++ b/xlators/mgmt/glusterd/src/glusterd-snapshot.c
@@ -6528,7 +6528,9 @@ glusterd_schedule_brick_snapshot (dict_t *dict, dict_t *rsp_dict,
         GF_ASSERT(dict);
         GF_ASSERT(snap);
 
-        synctask_barrier_init ((&args));
+        ret = synctask_barrier_init ((&args));
+        if (ret)
+                goto out;
         cds_list_for_each_entry (snap_vol, &snap->volumes, vol_list) {
                 volcount++;
                 brickcount = 0;
diff --git a/xlators/mgmt/glusterd/src/glusterd-syncop.c b/xlators/mgmt/glusterd/src/glusterd-syncop.c
index b63e61c..f051666 100644
--- a/xlators/mgmt/glusterd/src/glusterd-syncop.c
+++ b/xlators/mgmt/glusterd/src/glusterd-syncop.c
@@ -1226,7 +1226,9 @@ gd_lock_op_phase (glusterd_conf_t  *conf, glusterd_op_t op, dict_t *op_ctx,
         struct syncargs         args        = {0};
 
         this = THIS;
-        synctask_barrier_init((&args));
+        ret = synctask_barrier_init((&args));
+        if (ret)
+                goto out;
         peer_cnt = 0;
 
         rcu_read_lock ();
@@ -1359,7 +1361,9 @@ stage_done:
         }
 
         gd_syncargs_init (&args, aggr_dict);
-        synctask_barrier_init((&args));
+        ret = synctask_barrier_init((&args));
+        if (ret)
+                goto out;
         peer_cnt = 0;
 
         rcu_read_lock ();
@@ -1490,7 +1494,9 @@ commit_done:
         }
 
         gd_syncargs_init (&args, op_ctx);
-        synctask_barrier_init((&args));
+        ret = synctask_barrier_init((&args));
+        if (ret) 
+                goto out;
         peer_cnt = 0;
 
         rcu_read_lock ();
@@ -1566,7 +1572,9 @@ gd_unlock_op_phase (glusterd_conf_t  *conf, glusterd_op_t op, int *op_ret,
                 goto out;
         }
 
-        synctask_barrier_init((&args));
+        ret = synctask_barrier_init((&args));
+        if (ret)
+                goto out;
         peer_cnt = 0;
 
         if (cluster_lock) {
