commit 9597632e72d280716df3898ac1f2d512c415e812
Author: avonazhrets <avonazhrets@gmail.com>
Date:   Tue May 11 21:50:42 2021 +0300

    patch56

diff --git a/xlators/cluster/dht/src/dht-common.c b/xlators/cluster/dht/src/dht-common.c
index f521b9c..3d00726 100644
--- a/xlators/cluster/dht/src/dht-common.c
+++ b/xlators/cluster/dht/src/dht-common.c
@@ -755,6 +755,11 @@ dht_discover (call_frame_t *frame, xlator_t *this, loc_t *loc)
                 ret = dict_set_uint32 (local->xattr_req,
                                        conf->commithash_xattr_name,
                                        sizeof(uint32_t));
+                if (ret)
+                      gf_msg (this->name, GF_LOG_WARNING, 0,
+                        DHT_MSG_DICT_SET_FAILED,
+                        "%s: Failed to set dictionary value:key = %s",
+                        loc->path, conf->commithash_xattr_name);  
         }
 
         call_cnt        = conf->subvolume_cnt;
@@ -2542,6 +2547,10 @@ dht_lookup (call_frame_t *frame, xlator_t *this,
                 ret = dict_set_uint32 (local->xattr_req,
                                        conf->commithash_xattr_name,
                                        sizeof(uint32_t));
+                if (ret)
+                      gf_msg_debug (this->name, 0,
+                        "%s: Failed to set dictionary value:key = %s",
+                        loc->path, conf->commithash_xattr_name);  
         }
 
         if (!hashed_subvol)
