commit f930ac7fddf4f22fd192a611b9e4990aedeb351f
Author: avonazhrets <avonazhrets@gmail.com>
Date:   Tue May 11 16:43:07 2021 +0300

    patch53

diff --git a/cli/src/cli-rpc-ops.c b/cli/src/cli-rpc-ops.c
index 6b565c7..d02f9e0 100644
--- a/cli/src/cli-rpc-ops.c
+++ b/cli/src/cli-rpc-ops.c
@@ -6412,21 +6412,33 @@ cmd_profile_volume_brick_out (dict_t *dict, int count, int interval)
                 snprintf (key, sizeof (key), "%d-%d-%d-hits", count,
                           interval, i);
                 ret = dict_get_uint64 (dict, key, &profile_info[i].fop_hits);
+                if (ret) {
+                        gf_log("cli", GF_LOG_DEBUG, "failed to get %s from dict", key);
+                }
 
                 memset (key, 0, sizeof (key));
                 snprintf (key, sizeof (key), "%d-%d-%d-avglatency", count,
                           interval, i);
                 ret = dict_get_double (dict, key, &profile_info[i].avg_latency);
+                if (ret) {
+                        gf_log("cli", GF_LOG_DEBUG, "failed to get %s from dict", key);
+                }
 
                 memset (key, 0, sizeof (key));
                 snprintf (key, sizeof (key), "%d-%d-%d-minlatency", count,
                           interval, i);
                 ret = dict_get_double (dict, key, &profile_info[i].min_latency);
+                if (ret) {
+                        gf_log("cli", GF_LOG_DEBUG, "failed to get %s from dict", key);
+                }
 
                 memset (key, 0, sizeof (key));
                 snprintf (key, sizeof (key), "%d-%d-%d-maxlatency", count,
                           interval, i);
                 ret = dict_get_double (dict, key, &profile_info[i].max_latency);
+                if (ret) {
+                        gf_log("cli", GF_LOG_DEBUG, "failed to get %s from dict", key);
+                }
                 profile_info[i].fop_name = (char *)gf_fop_list[i];
 
                 total_percentage_latency +=
