commit 456af65e09fb0a8e66be1963165bef3c71a0f372
Author: avonazhrets <avonazhrets@gmail.com>
Date:   Sat May 15 13:47:26 2021 +0300

    patch79

diff --git a/xlators/performance/quick-read/src/quick-read.c b/xlators/performance/quick-read/src/quick-read.c
index 92b2f82..3821483 100644
--- a/xlators/performance/quick-read/src/quick-read.c
+++ b/xlators/performance/quick-read/src/quick-read.c
@@ -267,9 +267,10 @@ qr_content_extract (dict_t *xdata)
 {
 	data_t  *data = NULL;
 	void    *content = NULL;
+        int ret = 0;
 
-	dict_get_with_ref (xdata, GF_CONTENT_KEY, &data);
-	if (!data)
+	ret = dict_get_with_ref (xdata, GF_CONTENT_KEY, &data);
+	if (ret < 0 || !data)
 		return NULL;
 
 	content = GF_CALLOC (1, data->len, gf_qr_mt_content_t);
